addons/stock_picking_batch/models/stock_picking.py

```diff
@@ -86,3 +125,77 @@ def _should_show_transfers(self):
+                return new_batch
+
+        # If nothing came out of the loop, then no batch is doable given the conditions
+        return False
+
+    def _is_auto_batchable(self, picking):
+        """ Verifies if a picking can be put in a batch with another picking without violating auto_batch constrains.
+        """
+        res = True
+        if self.picking_type_id.batch_max_lines:
+            res = res and (len(self.move_ids) + len(picking.move_ids) <= self.picking_type_id.batch_max_lines)
+        if self.picking_type_id.batch_max_pickings:
+            # Sounds absurd. BUT if we put "batch max picking" to a value <= 1, makes sense ... Or not. Because then there is no point to batch.
+            res = res and self.picking_type_id.batch_max_pickings >= 2
+        return res
+
+    def _get_auto_batch_domain(self):
+        allowed_batch_states = ['draft', 'in_progress'] if self.picking_type_id.batch_auto_confirm else ['draft']
+        domain = [
+            ('id', '!=', self.id),
+            ('company_id', '=', self.company_id.id if self.company_id else False),
+            ('immediate_transfer', '=', False),
+            ('state', 'in', ('draft', 'waiting', 'confirmed', 'assigned')),  # TODO quwo : Might need to only allow assigned('ready') pickings
+            ('picking_type_id', '=', self.picking_type_id.id),
+            '|', ('batch_id', '=', False), ('batch_id.state', 'in', allowed_batch_states)
+        ]
+        if self.picking_type_id.batch_group_by_partner:
+            domain = expression.AND([domain, [('partner_id', '=', self.partner_id.id)]])
+        if self.picking_type_id.batch_group_by_destination:
+            domain = expression.AND([domain, [('partner_id.country_id', '=', self.partner_id.country_id.id)]])
+        if self.picking_type_id.batch_group_by_src_subloc and self.picking_type_id.default_location_src_id:
+            default_subloc_id = self._find_subloc_id_from_default(self.location_id, self.picking_type_id.default_location_src_id.child_ids.ids)
+            domain = expression.AND([domain, [('location_id', 'child_of', default_subloc_id)]])
+        if self.picking_type_id.batch_group_by_dest_subloc and self.picking_type_id.default_location_dest_id:
+            default_subloc_id = self._find_subloc_id_from_default(self.location_dest_id, self.picking_type_id.default_location_dest_id.child_ids.ids)
+            domain = expression.AND([domain, [('location_dest_id', 'child_of', default_subloc_id)]])
+
+        return domain
+
+    def _find_subloc_id_from_default(self, location, valid_subloc_ids):
+        """ Finds if a location is a child of a set of valid locations and if this is the case, returns which one.
+        """
+        if not valid_subloc_ids:
+            return False
+
+        while location:
+            if location.id in valid_subloc_ids:
+                return location.id
+            location = location.location_id
+        return False
```

Comment:
You could add that in stock.location object

Reply:
Done